Android/app/src/main/java/app/intra/sys/firebase/CountryCode.java

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2019 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.sys.firebase;
+
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import androidx.annotation.NonNull;
+import java.lang.reflect.Method;
+
+/**
+ * Static class for getting the system's country code.
+ */
+class CountryCode {
+  // The SIM or CDMA country.
+  private final @NonNull String deviceCountry;
+
+  // The country claimed by the attached cell network.
+  private final @NonNull String networkCountry;
+
+  CountryCode(Context context) {
+    TelephonyManager telephonyManager =
+        (TelephonyManager)context.getSystemService(Context.TELEPHONY_SERVICE);
+
+    deviceCountry = countryFromSim(telephonyManager);
+    networkCountry = telephonyManager.getNetworkCountryIso();
+  }
+
+  @NonNull String getDeviceCountry() {
+    return deviceCountry;
+  }
+
+  @NonNull String getNetworkCountry() {
+    return networkCountry;
+  }
+
+  /**
+   * @return The consensus country code, or "" if there isn't one.
+   */
+  @NonNull String getCombinedCountry() {
```

Comment:
I'm ok with providing a combined country, provided we know which country we are using. 
We need to better understand the data before trusting it. We need to see how those country parameters perform in the wild.

An alternative to my previous proposal is to add a new parameter that tells which country you picked. For example: COUNTRY_DEVICE, COUNTRY_NETWORK, COUNTRY_DEVICE_AND_NETWORK.

Though I have the impression the combined country will make it harder to pick different combinations. I suggested reporting the two locations instead (with the home or network country anonymized in case of roaming), because it's easy to apply filters like device_country == network_country. Or ip_country == network_country.

Reply:
OK, done.